test_mapbased_param/Cl_estimation.py

```diff
@@ -0,0 +1,375 @@
+        post_compsep_noise=hp.read_map(self.get_input('post_compsep_noise'),verbose=False, field=None)
+        frequency_maps=hp.read_map(self.get_input('frequency_maps'),verbose=False, field=None)
+        CMB_template_150GHz = hp.read_map(self.get_input('CMB_template_150GHz'), field=None)
+        
+        nhits_raw = hp.read_map(self.get_input('norm_hits_map'))
+        nhits = hp.ud_grade(nhits_raw,nside_out=self.config['nside'])
+        nh = mknm.get_mask(nhits, nside_out=self.config['nside'])
+
+        mask_patches = hp.read_map(self.get_input('mask_patches'), verbose=False, field=None)
+        noise_cov=hp.read_map(self.get_input('noise_cov'),verbose=False, field=None)
+        # reorganization of noise covariance 
+        instrument = {'frequencies':np.array(self.config['frequencies'])}
+        ind = 0
+        noise_cov_ = np.zeros((len(instrument['frequencies']), 3, frequency_maps.shape[-1]))
+        for f in range(len(instrument['frequencies'])) : 
+            for i in range(3): 
+                noise_cov_[f,i,:] = noise_cov[ind,:]*1.0
+                ind += 1
+        # removing I from all maps
+        noise_cov_ = noise_cov_[:,1:,:]
+
+
+        nside_map = hp.get_nside(clean_map[0])
+        print('nside_map = ', nside_map)
+        
+        w=nmt.NmtWorkspace()
+        b = binning_definition(self.config['nside'], lmin=self.config['lmin'], lmax=self.config['lmax'],\
+                                         nlb=self.config['nlb'], custom_bins=self.config['custom_bins'])
+
+        print('building mask ... ')
+        mask =  hp.read_map(self.get_input('binary_mask_cut'))
+        obs_pix = np.where(mask!=0)[0]
+        
+
+        if self.config['mask_apo'] != '':
+            mask_apo = hp.read_map(self.config['mask_apo'], verbose=False, field=None, h=False)
+            mask_apo = hp.ud_grade(mask_apo, nside_out=self.config['nside'])
+        else:
+            if self.config['extra_apodization']:
+                mask_apo = nmt.mask_apodization(mask, self.config['aposize'], apotype=self.config['apotype'])
+            else: mask_apo = mask*1.0
+
+            if ((self.config['noise_option']!='white_noise') 
+                    and (self.config['noise_option']!='no_noise')):
+                        # and (not self.config['no_inh'])):
+                nh = hp.smoothing(nh, fwhm=1*np.pi/180.0, verbose=False) 
+                nh /= nh.max()
+                mask_apo *= nh
+
+        fsky_eff = np.mean(mask_apo)
+        print('fsky_eff = ', fsky_eff)
+        np.savetxt(self.get_output('fsky_eff'), [fsky_eff])
+
+        print('building ell_eff ... ')
+        ell_eff = b.get_effective_ells()
+
+        #Read power spectrum and provide function to generate simulated skies
+        cltt,clee,clbb,clte = hp.read_cl(self.config['Cls_fiducial'])[:,:4000]
+        mp_t_sim,mp_q_sim,mp_u_sim=hp.synfast([cltt,clee,clbb,clte], nside=nside_map, new=True, verbose=False)
```

Comment:
Humm, why are you simulating anything here?

Reply:
If it's just to create a fake `NmtField` to compute the mode-coupling matrix, you could just make these `np.zeros` and save on `synfast`.